File: buff/classes/354/pub/sum/ml/sum.ml
(* ML sum program *)

let rec sum seq =
    match seq with
    | []   -> 0
    | h::t -> h+sum t;;

let rec sum1 seq =
    if seq=[] then
        0
    else
        List.hd(seq)+sum(List.tl(seq));;

let seq=[5; 6; 1; 8; 3; 7] in
print_int(sum seq);
print_newline();
print_int(sum1 seq);
print_newline();;

File: buff/classes/354/pub/sum/c-auto/sum.c
// C sum program

#include <stdio.h>

static int sum(int seq[], int n) {
  int s=0;
  for (int i=0; i<n; i++)
    s+=seq[i];
  return s;
}

int main() {
  auto int seq[]={5,6,1,8,3,7}; // stack allocation
  printf("%d\n",sum(seq,sizeof(seq)/sizeof(int)));
  return 0;
}
